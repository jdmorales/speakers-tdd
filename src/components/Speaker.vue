<template>
  <div>
    <img :src="speaker.image">
    <h2>{{speaker.username}}</h2>
  </div>
</template>

<script>
export default {
  props: {
    speaker: {
      type: Object,
      required: true
    }
  }
}
</script>

<style>

</style>