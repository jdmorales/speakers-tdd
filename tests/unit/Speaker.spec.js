import { shallowMount } from '@vue/test-utils'
import Speaker from '@/components/Speaker'

describe('Speaker', () => {
  let props
  let speaker = {
    username: '@myname',
    name: 'My Name',
    image:'https://myimage.com'
  }
  beforeEach(() => {
    props = {
      speaker
    }
  })

  const build = () => {
    const wrapper = shallowMount(Speaker, {
      // Add props while mounting
      propsData: props
    })

    return {
      wrapper
    }
  }

  it('renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the right', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.text()).toContain(speaker.username)
    expect(wrapper.find('img').attributes('src')).toBe(speaker.image)
  })
})
