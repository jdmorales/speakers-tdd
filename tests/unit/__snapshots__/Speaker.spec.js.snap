// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speaker renders the component 1`] = `
<div><img src="https://myimage.com">
  <h2>@myname</h2>
</div>
`;
